### ISSUE TITLE
reboot module fails with paramiko

### ISSUE THREAD

**Issue by barajus** [ansible/ansible#75589]
### Summary

When using Paramiko as transport, reboot fails.

### Issue Type

Bug Report

### Component Name

reboot

### Ansible Version

```console
ansible 2.9.19
same result with 2.10
```

### Configuration

```console
$ ansible-config dump --only-changed
cumulus@oob-mgmt-server:~/cumulus_ansible_modules$ ansible-config dump --only-changed
ANSIBLE_SSH_ARGS(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null
DEFAULT_CALLBACK_WHITELIST(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = [u'profile_tasks']
DEFAULT_FORCE_HANDLERS(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = True
DEFAULT_FORKS(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = 50
DEFAULT_JINJA2_EXTENSIONS(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = jinja2.ext.do
DEFAULT_MANAGED_STR(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = # Ansible Managed File
DEFAULT_ROLES_PATH(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = [u'/home/cumulus/cumulus_ansible_modules/roles']
DEFAULT_STDOUT_CALLBACK(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = yaml
DEFAULT_TRANSPORT(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = paramiko
DEPRECATION_WARNINGS(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = False
HOST_KEY_CHECKING(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = False
INTERPRETER_PYTHON(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = auto_silent
RETRY_FILES_ENABLED(/home/cumulus/cumulus_ansible_modules/ansible.cfg) = False
```

### OS / Environment

Ubuntu 18

### Steps to Reproduce

```yaml (paste below)
simple use reboot module with paramiko
```

### Expected Results

Server should reboot

### Actual Results

```console
RUNNING HANDLER [servers : restart networking] *******************************************************************************************************************************************************
task path: /home/cumulus/cumulus_ansible_modules/roles/servers/handlers/main.yml:2
Friday 27 August 2021  15:00:35 +0000 (0:00:00.216)       0:00:09.079 *********
<server01> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: cumulus on PORT 22 TO server01
<server01> EXEC /bin/sh -c 'echo ~cumulus && sleep 0'
<server01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/cumulus/.ansible/tmp `"&& mkdir "` echo /home/cumulus/.ansible/tmp/ansible-tmp-1630076435.67-4871-59861236764674 `" && echo ansible-tmp-1630076435.67-4871-59861236764674="` echo /home/cumulus/.ansible/tmp/ansible-tmp-1630076435.67-4871-59861236764674 `" ) && sleep 0'
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/system/setup.py
<server01> PUT /home/cumulus/.ansible/tmp/ansible-local-4731HKtJHv/tmpaGCSFk TO /home/cumulus/.ansible/tmp/ansible-tmp-1630076435.67-4871-59861236764674/AnsiballZ_setup.py
<server01> EXEC /bin/sh -c 'chmod u+x /home/cumulus/.ansible/tmp/ansible-tmp-1630076435.67-4871-59861236764674/ /home/cumulus/.ansible/tmp/ansible-tmp-1630076435.67-4871-59861236764674/AnsiballZ_setup.py && sleep 0'
<server01> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-ixqrdedmeysubrdfeanmksphdmrsmnhy ; /usr/bin/python /home/cumulus/.ansible/tmp/ansible-tmp-1630076435.67-4871-59861236764674/AnsiballZ_setup.py'"'"' && sleep 0'
<server01> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-byuftqzfqrqycoirruyuahkyiiivfkeb ; cat /proc/sys/kernel/random/boot_id'"'"' && sleep 0'
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/files/find.py
<server01> PUT /home/cumulus/.ansible/tmp/ansible-local-4731HKtJHv/tmpLTneAC TO /home/cumulus/.ansible/tmp/ansible-tmp-1630076435.67-4871-59861236764674/AnsiballZ_find.py
<server01> EXEC /bin/sh -c 'chmod u+x /home/cumulus/.ansible/tmp/ansible-tmp-1630076435.67-4871-59861236764674/ /home/cumulus/.ansible/tmp/ansible-tmp-1630076435.67-4871-59861236764674/AnsiballZ_find.py && sleep 0'
<server01> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-jdrxehpoqecufuadddjmmphaqlmmsaqh ; /usr/bin/python /home/cumulus/.ansible/tmp/ansible-tmp-1630076435.67-4871-59861236764674/AnsiballZ_find.py'"'"' && sleep 0'
reboot: rebooting server...
<server01> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-gwfiqnokixshsrmvbaxytgefzzrvlgic ; /sbin/shutdown -r 0 "Reboot initiated by Ansible"'"'"' && sleep 0'
reboot: validating reboot
reboot: attempting to get system boot time
reboot: attempting to get system boot time
reboot: attempting to get system boot time
reboot: attempting to get system boot time
reboot: attempting to get system boot time
[...]
reboot: attempting to get system boot time
reboot: attempting to get system boot time
reboot: attempting to get system boot time
fatal: [server01]: FAILED! =>
  msg: 'Failed to open session: SSH session not active'
META: ran handlers
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by barajus** [comment]
Hi @mkrizek , thanks for your answer.
With `devel`, the behavior is similar:
```
ansible [core 2.11.4]
  config file = /home/cumulus/cumulus_ansible_modules/ansible.cfg
  configured module search path = ['/home/cumulus/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/dist-packages/ansible
  ansible collection location = /home/cumulus/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.6.9 (default, Jan 26 2021, 15:33:00) [GCC 8.4.0]
  jinja version = 2.10
  libyaml = True
```



```
RUNNING HANDLER [servers : restart networking] *******************************************************************************************************************************************
task path: /home/cumulus/cumulus_ansible_modules/roles/servers/handlers/main.yml:2
<server01> ESTABLISH PARAMIKO SSH CONNECTION FOR USER: cumulus on PORT 22 TO server01
<server01> EXEC /bin/sh -c 'echo ~cumulus && sleep 0'
<server01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/cumulus/.ansible/tmp `"&& mkdir "` echo /home/cumulus/.ansible/tmp/ansible-tmp-1630171688.5731697-13980-71717739079431 `" && echo ansible-tmp-1630171688.5731697-13980-71717739079431="` echo /home/cumulus/.ansible/tmp/ansible-tmp-1630171688.5731697-13980-71717739079431 `" ) && sleep 0'
Using module file /usr/local/lib/python3.6/dist-packages/ansible/modules/setup.py
<server01> PUT /home/cumulus/.ansible/tmp/ansible-local-13950tb8gy9ef/tmpngpd38m0 TO /home/cumulus/.ansible/tmp/ansible-tmp-1630171688.5731697-13980-71717739079431/AnsiballZ_setup.py
<server01> EXEC /bin/sh -c 'chmod u+x /home/cumulus/.ansible/tmp/ansible-tmp-1630171688.5731697-13980-71717739079431/ /home/cumulus/.ansible/tmp/ansible-tmp-1630171688.5731697-13980-71717739079431/AnsiballZ_setup.py && sleep 0'
<server01> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-zbgutjusbtprbhxewisvqzxygozyfhwc ; /usr/bin/python /home/cumulus/.ansible/tmp/ansible-tmp-1630171688.5731697-13980-71717739079431/AnsiballZ_setup.py'"'"' && sleep 0'
<server01> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-nsrlquliapdsrznlggkprubztnwpesax ; cat /proc/sys/kernel/random/boot_id'"'"' && sleep 0'
Using module file /usr/local/lib/python3.6/dist-packages/ansible/modules/find.py
<server01> PUT /home/cumulus/.ansible/tmp/ansible-local-13950tb8gy9ef/tmpya0ba8cp TO /home/cumulus/.ansible/tmp/ansible-tmp-1630171688.5731697-13980-71717739079431/AnsiballZ_find.py
<server01> EXEC /bin/sh -c 'chmod u+x /home/cumulus/.ansible/tmp/ansible-tmp-1630171688.5731697-13980-71717739079431/ /home/cumulus/.ansible/tmp/ansible-tmp-1630171688.5731697-13980-71717739079431/AnsiballZ_find.py && sleep 0'
<server01> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-ydqbmlorhxqkbhwnitfxygnvzgptqyxd ; /usr/bin/python /home/cumulus/.ansible/tmp/ansible-tmp-1630171688.5731697-13980-71717739079431/AnsiballZ_find.py'"'"' && sleep 0'
reboot: rebooting server...
<server01> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-drnnlplehnbgysfgrqfmncclcsnsmpqj ; /sbin/shutdown -r 0 "Reboot initiated by Ansible"'"'"' && sleep 0'
reboot: validating reboot
reboot: attempting to get system boot time
reboot: attempting to get system boot time
reboot: attempting to get system boot time
[...]
fatal: [server01]: FAILED! => {
    "msg": "Failed to open session: SSH session not active"
}
```

**Comment by mkrizek** [comment]
Sorry, I meant the `devel` branch (where https://github.com/ansible/ansible/pull/74459 is present), so you need to install Ansible from git as documented in: https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#installing-and-running-the-devel-branch-from-source

**Comment by mkrizek** [comment]
I created a backport pull request https://github.com/ansible/ansible/pull/75614 for `2.11` release which still receives general bug fixes (as opposed to `2.9` and `2.10`, see https://docs.ansible.com/ansible/devel/reference_appendices/release_and_maintenance.html#ansible-core-release-cycle).

Since this has been fixed I am going to go ahead and close this.

If you have further questions please stop by IRC or the mailing list:
* IRC: #ansible on irc.libera.chat
* mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Comment by barajus** [comment]
Thanks :)
